packages/react/src/components/Modal/Modal.js

```diff
@@ -138,10 +141,13 @@ export default class Modal extends Component {
     size: PropTypes.oneOf(['xs', 'sm', 'lg']),
 
     /**
-     * Specify whether the modal should use 3rd party `focus-trap-react` for the focus-wrap feature.
-     * NOTE: by default this is true.
+     * Deprecated; Used for advanced focust-wrapping feature using 3rd party library,
+     * but it's now achieved without a 3rd party library.
      */
-    focusTrap: PropTypes.bool,
+    focusTrap: deprecate(
```

Comment:
Since we're deprecating this prop, are we not going to give the user the ability to choose whether or not they want the modal focus-wrapped?

Reply:
Thank you for asking @abbeyhrt - Focus-wrapping behavior runs regardless of `focusTrap` property, even before it was introduced or after it was introduced. Some time ago the codebase introduced a third-party [`focus-trap-react`](https://www.npmjs.com/package/focus-trap-react) library for better focus-wrapping behavior, but it introduced lots of side effects as the PR description of this PR links to. `focusTrap` property was introduced to disable such third-party library to avoid the side effects. This PR removes the property because the third-party is no longer needed for focus-wrapping.